common/workunit/workunit.cpp

```diff
@@ -14024,23 +14024,52 @@ bool executeGraphOnLingeringThor(IConstWorkUnit &workunit, const char *graphName
 
 static void setResources(StringBuffer &jobYaml, const IConstWorkUnit *workunit, const char *process)
 {
+    class GenerateResources
+    {
+    public:
+        GenerateResources(StringBuffer &_jobYaml, IPropertyTree *_cconfig): jobYaml(_jobYaml), cconfig(_cconfig) {};
+
+        void setResourceValue(const char *category, const char *resourceName,  unsigned wuvalue, const char *units)
+        {
+            if (!wuvalue && !cconfig)
+                return;
+            StringBuffer tag2replace;
+            tag2replace.setf("#%s-%s", category, resourceName);
+
+            StringBuffer s;
+            s.clear().appendf("%s: ", resourceName);
+            if (wuvalue)
+            {
+                jobYaml.replaceString(tag2replace, s.appendf("\"%u%s\"", wuvalue, units));
+            }
+            else
+            {
+                StringBuffer p;
+                p.appendf("./%s/@%s", category, resourceName);
+                const char *v = cconfig->queryProp(p.str());
+                if (v)
+                    jobYaml.replaceString(tag2replace, s.appendf("\"%s\"", v));
+            }
+        }
+    private:
+        StringBuffer &jobYaml;
+        Owned<IPropertyTree> cconfig;
+    } generateResources (jobYaml, queryComponentConfig().getPropTree("childContainersResources"));
```

Comment:
I find the way this is coded very confusing. The fact that the calls to setResourceValue also use the config as an input parameter and modify jobYaml is hidden from all the calls...